crypto/evp/digest.c

```diff
@@ -303,26 +162,27 @@ int EVP_DigestUpdate(EVP_MD_CTX *ctx, const void *data, size_t count)
         return 0;
     }
 
-    if (ctx->digest == NULL
-            || ctx->digest->prov == NULL
-            || (ctx->flags & EVP_MD_CTX_FLAG_NO_INIT) != 0)
-        goto legacy;
+#ifndef FIPS_MODE
+    {
+        int ret = 0;
+
+        if (legacy_evp_digest_update(ctx, data, count, &ret))
```

Comment:
Similarly, one might look at this as an alternative:

``` C
        if (legacy_evp_digest_is_legacy(ctx))
            return legacy_evp_digest_update(ctx, data, count);
```

Reply:
... and similarly for the final call, for ciphers, etc...

Reply:
Even this simple one doesnt quite work... What if ctx->digest is NULL?

    if (ctx->digest == NULL)
        goto err;
    if (ctx->digest->prov != NULL)